t/50proxy-forward-close-connection.t

```diff
@@ -11,12 +11,12 @@ plan skip_all => 'plackup not found'
 plan skip_all => 'Starlet not found'
     unless system('perl -MStarlet /dev/null > /dev/null 2>&1') == 0;
 
-my $keep_alive_upstream_port = empty_port();
-my $close_conn_upstream_port = empty_port();
 my $curl = 'curl --silent --dump-header /dev/stderr';
 
 subtest 'basic' => sub {
     my $doit = sub {
+    my $keep_alive_upstream_port = empty_port();
+    my $close_conn_upstream_port = empty_port();
```

Comment:
Thank you for the PR.

I am not sure if the proposed change is to going to cause any difference.
* The last element returned by `one_shot_http_upstream` is a destructor (a `Scope::Guard` object) that kills the child process listening on a given port and waits for that process to die.
* This element is assigned to `$g1` and `$g2` in the `$doit` subroutine.
* At the moment of the subroutine exit, `$g1` and `$g2` goes out of scope and their destructor will be called.

I tend to believe that what we saw here is a TOCTOU issue that would be addressed by approaches like https://github.com/h2o/h2o/pull/3237. WDYT?

Reply:
Ah, good point.  `one_shot_http_upstream` failed to bind the first time around.

It was not a problem on the second `$doit` call.  `one_shot_http_upstream`'s listener uses `SO_REUSEADDR` (`Reuse => 1` in the `IO::Socket::INET->new` call), so it should not have a problem opening the same listener again in the second `$doit` call.

Reply:
That makes sense, thank you for pushing #3237 !